client/modules/IDE/actions/files.js

```diff
@@ -156,6 +156,25 @@ export function updateFileName(id, name) {
   };
 }
 
+export function saveFile(id) {
+  return (dispatch, getState) => {
+    const state = getState();
+    const UpdatedFile = state.files.find(file => file.id === id);
+    if (UpdatedFile && state.project.id) {
+      axios.put(`${ROOT_URL}/projects/${state.project.id}/files/${id}`, UpdatedFile, { withCredentials: true })
+        .then(() => {
+          dispatch(setUnsavedChanges(false));
```

Comment:
There is a new issue with a dot appearing after file name that used to indicate unsaved changes. File will be saved after renaming file thus dot should have to disappear. This behaviour controlled by setUnsavedChanges()`. But this function is used for all files in project. If  I don't use it here there are two problems, one with a dot and second is warning message show when leaving site. When use of it cause problem for other unsaved files.

Reply:
yeah, i've noticed this! it's been super annoying. i'm not sure what's going on here. i may revert a commit that seems to have made this worse.